opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -581,21 +581,25 @@ static int mca_btl_ofi_init_device(struct fi_info *info)
 fail:
     /* clean up */
 
+    if (NULL != ep && !module->is_scalable_ep) {
+        fi_close(&ep->fid);
```

Comment:
Looks like this could be double closed on line 596?

Reply:
I added a comment about checking for NULL to avoid a double-close. If the endpoint is scalable, this will not be a double-close. If the endpoint is basic, then we needed to close it before closing anything else.

Please let me know if I should be less succinct in comments.

Reply:
The comments look good to me, thanks.

Couldn't `fi_close(&ep->fid);` be called again on line 603 though still? Should ep be set to NULL here? (or that line on 603 removed?)

Reply:
Good catch. Thank you. I will set ep to NULL.

Reply:
I've made the change and pushed it .
Before pushing it, I tested with both a scalable and a basic provider, forcing it into the fail code by using gdb to set a ret to a non-zero value in mca_btl_ofi_init_device and putting a breakpoint on fi_close to verify that without the fix there will be a double-close but with the fix, fi_close functions properly during the fail.